rasa/core/events/__init__.py

```diff
@@ -347,13 +348,8 @@ def message(self) -> Dict[Text, Any]:
 
         m = self.data.copy()
         m["text"] = self.text
+        m.update(self.metadata.copy())
 
-        if m.get("image") == m.get("attachment"):
```

Comment:
why did you remove this? removing this should break the tests

Reply:
In the current method, if there's no attachment, then image gets put as both image and attachment, and then we delete the image from attachment here. 

I believe I took care of this [here](https://github.com/RasaHQ/rasa/pull/3423/files#diff-d2d909f773bca52743b6b9106d3184a5R135) -- If there's no attachment and an image then the image will get popped into image and therefore not put into attachment.

Reply:
ok i see, that won't work: for legacy reasons we *want* to set the image as the attachement (having the same value for both if there is no other attachament).  Let's keep this part here and change the code you linked to
```
            "elements": message.pop("elements", None),
            "quick_replies": message.pop("quick_replies", None),
            "buttons": message.pop("buttons", None),
            # for legacy / compatibility reasons we need to set the image to be the attachement if there
            # is no other attachement (the `.get` is intentional - no `pop` as we still need the `image` 
            # property to set it in the following line)
            "attachment": message.pop("attachment", None) or message.get("image", None),
            "image": message.pop("image", None),
            "custom": message.pop("custom", None),
```

Reply:
right, but why are we setting the image as the attachment just to get rid of it later? is it just bc there's a processing step in between that uses attachment that i'm missing? it didn't seem to break the tests

Reply:
also happy to do it this way and come back to re-evaluate later, just feel like there's a cleaner way to do this. maybe we `setdefault()` to image and then update with attachment? This would make it so that 

> image as the attachement (having the same value for both if there is no other attachament).

Reply:
it is not about any code in here, but the `BotUttered` event gets persisted and then used e.g. in the chatui frontend on the webpage which also reads these events. Changing the value of the attachement would break that ui.

Reply:
this is also why we actualy need to set the value and not just pretend it has this value (using `setdefault`)

Reply:
Then doesn't moving the image to `image` instead of `attachment` below also break the BotUttered event?

Reply:
Then doesn't moving the image to `image` instead of `attachment` below also break the BotUttered event?

Reply:
Also can't we handle this in the `create_bot_utterance()` method?

Reply:
I am not sure what you mean.

- `create_bot_utterance` needs to create a backwards compatible `BotUttered` event (setting image as attachement if no other attachement is present) 
- this `message` function here (in the lines we are currently commenting on) is used to create the message dict that gets handed over to the output channel. to avoid sending the image twice, we remove it from the attachement if it is set there.

Reply:
Ahhhhhh okay. now i'm with it, this is what I meant by this 

> is it just bc there's a processing step in between that uses attachment that i'm missing?

so the persisting of the botmessage needs the attachment to also be the image, but then we delete it before sending it to the output channel. All is cleared up! Will clean this up, let me know if you want to rereview before merge but i'll jsut implement your suggestions

Reply:
just send me a quick note once you are ready and I'll hit the merge button :+1: